providers/azure/aci.go

```diff
@@ -724,11 +724,50 @@ func (p *ACIProvider) getContainers(pod *v1.Pod) ([]aci.Container, error) {
 			}
 		}
 
+		if container.LivenessProbe != nil {
+			c.LivenessProbe = getProbe(container.LivenessProbe)
+		}
+
+		if container.ReadinessProbe != nil {
+			c.ReadinessProbe = getProbe(container.ReadinessProbe)
+		}
+
 		containers = append(containers, c)
 	}
 	return containers, nil
 }
 
+func getProbe(probe *v1.Probe) *aci.ContainerProbe {
+	// Probes have can have a Exec or HTTP Get Handler.
+	// Create those if they exist, then add to the
+	// ContainerProbe struct
+	var exec *aci.ContainerExecProbe
+	if probe.Handler.Exec != nil {
+		exec = &aci.ContainerExecProbe{
+			Command: probe.Handler.Exec.Command,
+		}
+	}
+
+	var httpGET *aci.ContainerHTTPGetProbe
+	if probe.Handler.HTTPGet != nil {
+		httpGET = &aci.ContainerHTTPGetProbe{
+			Port:   probe.Handler.HTTPGet.Port.IntValue(),
+			Path:   probe.Handler.HTTPGet.Path,
+			Scheme: string(probe.Handler.HTTPGet.Scheme),
+		}
+	}
+
+	return &aci.ContainerProbe{
```

Comment:
ACI requires that 1 and only 1 of Exec and HttpGet is specified in the probe. We'd better do the same validation here.

Reply:
K8s has that restriction as well:

```
The Pod "nginx" is invalid: spec.containers[0].livenessProbe.httpGet: Forbidden: may not specify more than 1 handler type
```

I figured that would cover this, but I'll add validation to be safe!